controllers/webhook.js

```diff
@@ -0,0 +1,260 @@
+            }
+          }
+
+          if (reqAuthenticated) {
+            await _handleMergeRequest(version, service, username, repository, branch, req.body).catch((errors) => {
+              errorsRaised = errors
+            })
+          }
+        }
+      }
+
+      break
+    default:
+      errorsRaised.push('Unexpected service specified.')
+  }
+
+  if (errorsRaised.length > 0) {
+    res.status(400).send({
+      errors: JSON.stringify(errorsRaised)
+    })
+  } else {
+    res.status(200).send({
+      success: true
+    })
+  }
+}
+
+const _verifyGitHubSignature = function (secret, data, signature) {
+  const signedData = 'sha1=' + crypto.createHmac('sha1', secret).update(data).digest('hex')
+  return bufferEq(Buffer.from(signature), Buffer.from(signedData))
+}
+
+const _handleMergeRequest = async function (version, service, username, repository, branch, data) {
+  // Allow for multiple errors to be raised and reported back.
+  const errors = []
+
+  const ua = config.get('analytics.uaTrackingId')
+    ? require('universal-analytics')(config.get('analytics.uaTrackingId'))
+    : null
+
+  let gitService = null
+  let mergeReqNbr = null
+  if (service === 'github') {
+    /*
+     * In v1 of the endpoint, the service, username, repository, and branch parameters were
+     * ommitted. As such, if not provided in the webhook request URL, pull them from the webhook
+     * payload.
+     */
+    if (username === null || typeof username === 'undefined') {
+      username = data.repository.owner.login
+    }
+    if (repository === null || typeof repository === 'undefined') {
+      repository = data.repository.name
+    }
+    if (branch === null || typeof branch === 'undefined') {
+      branch = data.pull_request.base.ref
+    }
+
+    gitService = await gitFactory.create('github', {
+      version: version,
+      username: username,
+      repository: repository,
+      branch: branch
+    })
+    mergeReqNbr = data.number
+  } else if (service === 'gitlab') {
+    gitService = await gitFactory.create('gitlab', {
+      version: version,
+      username: username,
+      repository: repository,
+      branch: branch
+    })
+    mergeReqNbr = data.object_attributes.iid
+  } else {
+    errors.push('Unable to determine service.')
+    return Promise.reject(errors)
+  }
+
+  if (mergeReqNbr === null || typeof mergeReqNbr === 'undefined') {
+    errors.push('No pull/merge request number found.')
+    return Promise.reject(errors)
+  }
+
+  let review = await gitService.getReview(mergeReqNbr).catch((error) => {
+    const msg = `Failed to retrieve merge request ${mergeReqNbr} - ${error}`
+    console.error(msg)
+    errors.push(msg)
+    return Promise.reject(errors)
+  })
+
+  /*
+   * We might receive "real" (non-bot) pull requests for files other than Staticman-processed
+   * comments. Ignore these by filtering on Staticman-created merge request branches.
+   */
+  if (review.sourceBranch.indexOf('staticman_') > -1) {
```

Comment:
```suggestion
  /*
   * We might receive "real" (non-bot) pull requests for files other than Staticman-processed
   * comments. Ignore these by filtering on Staticman-created merge request branches.
   */
  if (review.sourceBranch.indexOf('staticman_') > -1) {
```

Comment could be helper function like `_isStaticmanComment(review)`

Reply:
Changed as requested.